Utilities/StorageFactory/src/RemoteFile.cc

```diff
@@ -52,27 +54,30 @@ int RemoteFile::local(const std::string &tmpdir, std::string &temp) {
   // This is better for grid jobs as the current directory is
   // likely to have more space, and is more optimised for
   // large files, and is cleaned up after the job.
-  if (tmpdir.empty() || tmpdir == ".") {
-    size_t len = pathconf(".", _PC_PATH_MAX);
-    char *buf = (char *)malloc(len);
-    getcwd(buf, len);
+  std::vector<char> temp_chars;
 
-    temp.reserve(len + 32);
-    temp = buf;
-    free(buf);
+  if (tmpdir.empty() || tmpdir == ".") {
+    std::filesystem::path current_path = std::filesystem::current_path();
+    temp_chars.assign(current_path.string().begin(), current_path.string().end());
```

Comment:
The returned from `current_path.string()` is a temporary so calling it twice gives two different strings which means the begin/end are not proper pairs. Please do

```suggestion
    auto spath = std::filesystem::current_path().string();
    temp_chars.reserve(spath.size()+30);
    temp_chars.assign(spath.begin(), spath.end());
```